### PR TITLE
Multi-point multiplication support

### PR DESCRIPTION
**Pull request by sipa**
This uses the algorithm (Strauss with wNAF) from #464 to implement multi-multiplication, together with the API, tests, and scratch space support from #461.

### CODE REVIEW

File: src/tests.c [hunk 1]
```diff
@@ -2487,6 +2522,243 @@ void run_ecmult_const_tests(void) {
...
+        random_scalar_order(&sc[1]);
+
+        random_group_element_test(&ptg);
+        secp256k1_gej_set_ge(&ptgj, &ptg);
+        pt[0] = ptg;
+        pt[1] = secp256k1_ge_const_g;
+
+        /* 1-point */
+        secp256k1_ecmult(&ctx->ecmult_ctx, &r2, &ptgj, &sc[0], &szero);
+        secp256k1_ecmult_multi(&ctx->ecmult_ctx, scratch, &ctx->error_callback, &r, &szero, ecmult_multi_callback, &data, 1);
```
Comment (jonasnick) [inline src/tests.c]: The tests should check the return value of secp256k1_ecmult_multi

**Comment by sipa** [comment]
Added an ecmult_multi benchmark tool.

File: src/bench_ecmult.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        secp256k1_scalar_negate(&total, &total);
+        secp256k1_ecmult(&data->ctx->ecmult_ctx, &data->expected_output[iter], NULL, &zero, &total);
+    }
+
+    /* Run the benchmark. */
+    sprintf(str, includes_g ? "ecmult_%ig" : "ecmult_%i", (int)count);
+    run_benchmark(str, bench_ecmult, bench_ecmult_setup, bench_ecmult_teardown, data, 10, count * (1 + ITERS / count));
+}
+
+int main(int argc, char **argv) {
```
Comment (jonasnick) [inline src/bench_ecmult.c]: Unused arguments

**Comment by apoelstra** [comment]
superceded by #486

**Pull request closed without merging**